pkg/raft/opts.go

```diff
@@ -257,3 +289,59 @@ func StatsDAddr(statsdAddr string) Option {
 		return nil
 	}
 }
+
+func (this *options) MaxSnapFiles() uint {
+	return uint(this.maxSnapFiles)
+}
+
+func (this *options) MaxWALFiles() uint {
+	return uint(this.maxWALFiles)
+}
+
+func (this *options) SnapshotCount() uint64 {
+	return uint64(this.snapshotCount)
+}
+
+func (this *options) SnapshotCatchUpEntries() uint64 {
+	return uint64(this.snapshotCatchUpEntries)
+}
+
+func MaxSnapFiles(count int) Option {
+	return func(opts *options) error {
+		if count < 0 {
+			return errors.New("maxSnapFiles cannot be negative")
+		}
+		opts.maxSnapFiles = count
+		return nil
+	}
+}
+
+func MaxWALFiles(count int) Option {
+	return func(opts *options) error {
+		if count < 0 {
+			return errors.New("maxWALFiles cannot be negative")
+		}
+		opts.maxWALFiles = count
+		return nil
+	}
+}
+
+func SnapshotCount(count int64) Option {
+	return func(opts *options) error {
+		if count < 0 {
+			return errors.New("snapshotCount cannot be negative")
+		}
+		opts.snapshotCount = count
+		return nil
+	}
+}
+
+func SnapshotCatchUpEntries(count int64) Option {
+	return func(opts *options) error {
+		if count < 0 {
```

Comment:
Check if we should disallow `count == 0` as well ?

Reply:
Done